tests/standalone_tests/test_sql_injection_guard.py

```diff
@@ -0,0 +1,486 @@
+            {},
+            id="optional input not provided",
+        ),
+        pytest.param(
+            dummy_method_with_one_optional_input,
+            ["alnum1"],
+            {},
+            id="optional input provided",
+        ),
+        pytest.param(
+            dummy_method_with_one_input,
+            [5],
+            {},
+            id="non string passed",
+        ),
+        pytest.param(
+            dummy_method_with_one_input,
+            [[5, 6, 7, "alnum1"]],
+            {},
+            id="list of ints passed",
+        ),
+        pytest.param(
+            dummy_method_with_one_input,
+            [{"x": "alnum1"}],
+            {},
+            id="dict passed",
+        ),
+        pytest.param(
+            dummy_method_with_one_input,
+            [DummyClass()],
+            {},
+            id="object is not being validated",
+        ),
+        pytest.param(
+            dummy_method_with_one_input,
+            [2],
+            {},
+            id="integer is not being validated",
+        ),
+        pytest.param(
+            dummy_method_with_one_input,
+            [["alnum1", "alnum2"]],
+            {},
+            id="str passed inside list",
+        ),
+        pytest.param(
+            dummy_method_with_one_input,
+            [{"alnum1": "alnum2"}],
+            {},
+            id="str passed inside dict",
+        ),
+        pytest.param(
+            dummy_method_with_one_input,
+            [[2, 3, 4, 5]],
+            {},
+            id="list of integers is not being validated",
+        ),
+        pytest.param(
+            dummy_method_with_filter_input,
+            [
+                {
+                    "condition": "OR",
+                    "rules": [
+                        {
+                            "id": "test_age_value",
+                            "field": "test_age_value",
+                            "type": "int",
+                            "input": "number",
+                            "operator": "equal",
+                            "value": 17,
+                        },
+                        {
+                            "id": "test_pupil_reactivity_right_eye_result",
+                            "field": "test_pupil_reactivity_right_eye_result",
+                            "type": "string",
+                            "input": "text",
+                            "operator": "not_equal",
+                            "value": "Nonreactive",
+                        },
+                    ],
+                    "valid": True,
+                }
+            ],
+            {},
+            id="filters are being validated",
+        ),
+    ]
+
+
+@pytest.mark.parametrize("func, args, kwargs", get_parametrization_list_success_cases())
+def test_sql_injection_guard(func, args, kwargs):
+    try:
+        func(*args, **kwargs)
+    except Exception as exc:
+        pytest.fail(f"No exception should be raised. Exception: {exc}.")
```

Comment:
Oh, the parametrized tests again! 

I need to check in three places in order to understand one test: in `test_sql_injection_guard`, in the parameter and in the corresponding dummy method. Once I visit each one of them I already forgot which parameter I just saw and which one is next.

What's wrong with writing readable, self contained tests? Like:
```python
def test_one_input():

    @sql_injection_guard("input", isalnum)
    def dummy_method(input):
        return input

    try:
        dummy_method("alnum1")
    except Exception as exc:
        pytest.fail(f"No exception should be raised. Exception: {exc}.")
```

I am sorry, I cannot review these tests.